packages/core/src/modal/modal.stories.ts

```diff
@@ -44,7 +44,7 @@ export function API(args: any) {
       <cds-button status="primary" type="button" @click=${showApiModal}>Show Modal</cds-button>
       <cds-modal ...="${spreadProps(getElementStorybookArgs(args))}" id="${modalId}">
         <cds-modal-header>
-          <h3 cds-text="title">${args['cds-modal-header']}</h3>
+          <h3 cds-text="title" cds-first-focus tabindex="-1">${args['cds-modal-header']}</h3>
```

Comment:
Instead of requiring the tabindex could the focus trap add it if needed? Maybe something like this?

```typescript
    const firstFocusElement = fte.querySelector('[cds-first-focus]');

    if (firstFocusElement && !firstFocusElement.hasAttribute('tabindex')) {
      firstFocusElement.setAttribute('tabindex', '-1');
    }
```
https://github.com/vmware/clarity/blob/next/packages/core/src/internal/utils/focus-trap.ts#L142

Reply:
The tabindex is only required if the element is not focuasable. Assigning tabindex of `-1` to, say, a button would be a bug.

I think this is an interesting suggestion. But I feel like it is out of scope of the ask in this issue. I'm happy to open another ticket to look into that focustrap feature or discuss offline.